components/cloud/src/blob.rs

```diff
@@ -133,6 +133,12 @@ impl std::ops::Deref for StringNonEmpty {
     }
 }
 
+impl std::ops::DerefMut for StringNonEmpty {
```

Comment:
Generally this isn't safe for this type. One can always set a StringNonEmpty to an empty string by this mutable reference.

Consider remove this (as this may not be used.) If you think this is nesseary, I will perfer providing some method instead of directly give the caller the internal string.